src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -154,3 +199,65 @@ std::optional<Point> CreaseDefenderFSM::findDefenseAreaIntersection(
 
     return std::nullopt;
 }
+
+bool CreaseDefenderFSM::ballNearbyWithoutThreat(const Update& event)
+{
+    Point robot_position = event.common.robot.position();
+    Point ball_position  = event.common.world_ptr->ball().position();
+
+    std::optional<Robot> nearest_friendly_to_ball =
+        event.common.world_ptr->friendlyTeam().getNearestRobot(ball_position);
+    std::optional<Robot> nearest_enemy_to_ball =
+        event.common.world_ptr->enemyTeam().getNearestRobot(robot_position);
```

Comment:
This is a bit confusing. Based on your explanation above, shouldn't this be
```suggestion
    std::optional<Robot> nearest_enemy_to_ball =
        event.common.world_ptr->enemyTeam().getNearestRobot(ball_position);
```
As in the robot that's closest to the ball and poses the highest risk of taking possession of the ball?

Reply:
good catch